cpp/src/arrow/util/utf8_util_benchmark.cc

```diff
@@ -70,16 +70,44 @@ static void BenchmarkUTF8Validation(
   state.SetBytesProcessed(state.iterations() * s.size());
 }
 
+static void BenchmarkASCIIValidation(
+    benchmark::State& state,  // NOLINT non-const reference
+    const std::string& s, bool expected) {
+  auto data = reinterpret_cast<const uint8_t*>(s.data());
+  auto data_size = static_cast<int64_t>(s.size());
+
+  InitializeUTF8();
+#if defined(ARROW_HAVE_NEON) || defined(ARROW_HAVE_SSE4_2)
```

Comment:
How about calling a single function `ValidateAscii` at caller side? Then, the function calls simd or non-simd function based on ifdef. It makes code simple by reducing the number of the repetition `#if defined(ARROW_HAVE_NEON) || defined(ARROW_HAVE_SSE4_2) ...`.

[Here](https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/byte_stream_split.h#L614-L622) is an example.

Reply:
Yup, it makes sense. 
The PR was updated, thanks for your comments.